### PR TITLE
Add report to view updates on endpoints

### PR DESCRIPTION
**Pull request by kena-SL**
This report gets a list of endpoints that have been updated
Considers most recent resources to get the difference in entity count

### CODE REVIEW

File: service_report/Identify_updated_endpoints.ipynb [hunk 1]
```diff
@@ -0,0 +1,276 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d918690d-07cf-4ee6-abdc-a6f85b4faa71",
+   "metadata": {},
+   "source": [
+    "This Report gets a list of endpoints that have updated along with the entity count."
```
Comment (amiriam-scottlogic) [inline service_report/Identify_updated_endpoints.ipynb:8]: I think there was talk of a template description for Jupyter Notebooks, but I'm not sure!

**Review by amiriam-scottlogic — CHANGES_REQUESTED** [review]

File: service_report/Identify_updated_endpoints.ipynb [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    }
+   ],
+   "source": [
+    "datasette_url = \"https://datasette.planning.data.gov.uk/\"\n",
+    "result_df = pd.DataFrame()\n",
+    "\n",
+    "def get_all_endpoints(dataset):\n",
+    "    global result_df\n",
+    "    params = urllib.parse.urlencode({\n",
+    "        \"sql\": f\"\"\"\n",
```
[the remark this answers is not in the document]
Comment (kena-SL) [reply service_report/Identify_updated_endpoints.ipynb:51]: It is how Github looks at .ipynb files
Here it looks okay I think ?
https://github.com/digital-land/jupyter-analysis/blob/Identify-updated-endpoints/service_report/Identify_updated_endpoints.ipynb

File: service_report/Identify_updated_endpoints.ipynb [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    "        select t1.endpoint,\n",
+    "        t3.pipeline from(select endpoint from resource_endpoint \n",
+    "        group by endpoint having count(endpoint)>1) t1 \n",
```
[the remark this answers is not in the document]
Comment (kena-SL) [reply service_report/Identify_updated_endpoints.ipynb:54]: A new resource gets created when there is change on the endpoint so If an endpoint has only one resource then it doesn't have any updates

File: service_report/Identify_updated_endpoints.ipynb [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    "        count = pd.read_csv(url)\n",
+    "        per_resource_count.append(count.iloc[0, 0])\n",
+    "    all_resource_count[key] = per_resource_count\n",
+    "\n",
+    "endpoints_with_diff_count={}\n",
+    "for key,value in all_resource_count.items():\n",
+    "    flag = True if value[0] != value[1] else False\n",
+    "    if flag:\n",
+    "        diff=value[0]-value[1]\n",
+    "        if diff < 0:\n",
```
[the remark this answers is not in the document]
Comment (kena-SL) [reply service_report/Identify_updated_endpoints.ipynb:188]: This one just checks for change in entity count
If I am not wrong the pipeline does handle updates made on existing entities on its own but can be something to investigate

File: service_report/Identify_updated_endpoints.ipynb (hunk 1 above)
Comment (kena-SL) [reply service_report/Identify_updated_endpoints.ipynb:8]: Oh yes
Thanks for reminding!

File: service_report/Identify_updated_endpoints.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (amiriam-scottlogic) [reply service_report/Identify_updated_endpoints.ipynb:195]: I'm unsure whether it does - it's worth checking! If it does, great! If it doesn't, I guess we can see if there's an easy solution for this PR - if there isn't and it's a big job, worth making a separate ticket

[the remark this answers is not in the document]
Comment (kena-SL) [reply service_report/Identify_updated_endpoints.ipynb:195]: The requirement this report fulfils is seeing if an endpoint had any entities added or deleted and that is the scope of this report. Looking at the recent resources from an endpoint and its entity counts is one way of doing this.
We can prioritise the other bit of work you mentioned if needed.

**Review by amiriam-scottlogic — APPROVED** [review]

**Pull request merged**